File: python/rotational-cipher/rotational_cipher.py
def rotate(text, key):
    def rot(c):
        if c.isalpha():
            nc = ord(c) + key
            if c.isupper() and nc > ord("Z") or nc > ord("z"):
                nc -= 26
            return chr(nc)
        return c
    return "".join(map(rot, list(text)))

File: zig/rotational-cipher/rotational_cipher.zig
const std = @import("std");

pub fn rotate(allocator: std.mem.Allocator, text: []const u8, shiftKey: u5) std.mem.Allocator.Error![]u8 {
    const cypher = try allocator.dupe(u8, text);
    for (cypher) |*c| {
        if (std.ascii.isAlphabetic(c.*)) {
            const first: u8 = if (std.ascii.isUpper(c.*)) 'A' else 'a';
            c.* = (c.* - first + shiftKey) % 26 + first;
        }
    }
    return cypher;
}
